katalog/strimzi-cluster-operator/MAINTENANCE.md

```diff
@@ -23,6 +23,107 @@ repository. The patching is done via `kustomization.yaml` and `patch-images.yaml
+
+5. Change the ```spec.kafka.version``` of the Kafka resource to ```3.8.0```.
+   
+   > Example: update ```spec.kafka.version``` of ```kafka-cluster.yaml``` in ```examples/kafka-standard```.
+
+   Apply the changes:
+
+   ```kustomize build | kubectl apply -f -```
+
+   The previous error ```Exception: Unsupported Kafka.spec.kafka.version: 3.2.3. Supported versions are: [3.7.0, 3.7.1, 3.8.0]``` at this point will disappear and the ```strimzi-cluster-operator``` will start the reconciliation with the Kafka resource.
+
+   The update of ```zookeeper```, ```kafka```, ```exporter``` and ```entity-operator``` pods will be automatically triggered after the version update.
+
+6. Wait for *two* rollouts of the Kafka resources to be completed, and check for their status.
+
+   ```bash
+   ~ % kubectl rollout status deploy sighup-kafka-cluster-kafka-exporter -n sighup
+   deployment "sighup-kafka-cluster-kafka-exporter" successfully rolled out
+   ```
+
+   ```bash
+   ~ % kubectl rollout status deploy sighup-kafka-cluster-entity-operator -n sighup
+   deployment "sighup-kafka-cluster-entity-operator" successfully rolled out
+   ```
+
+   ```bash
+   ~ % kubectl get strimzipodsets -n sighup
+   NAME                             PODS   READY PODS   CURRENT PODS   AGE
+   sighup-kafka-cluster-kafka       3      3            3              1h
+   sighup-kafka-cluster-zookeeper   3      3            3              1h
+   ```
+
+7. Check the image used by ```zookeeper```, ```kafka```, ```exporter``` and ```entity-operator``` pods.  
+
+
+   ```zookeeper```, ```kafka```, and ```exporter```pods should now use the image ```registry.sighup.io/fury/strimzi/kafka:0.43.0-kafka-3.8.0```. 
+
+   ```shell
+   kubectl get pod sighup-kafka-cluster-kafka-* -n sighup -o jsonpath='{.spec.containers[0].image}'
+
+   kubectl get pod sighup-kafka-cluster-kafka-exporter-* -n sighup -o jsonpath='{.spec.containers[0].image}'
+
+   kubectl get pod sighup-kafka-cluster-zookeeper-* -n sighup -o jsonpath='{.spec.containers[0].image}'
+   ``` 
+
+    The ```entity-operator``` pod should use the image ```registry.sighup.io/fury/strimzi/operator:0.43.0```.
+
+    ```shell
+    kubectl get pod sighup-kafka-cluster-entity-operator-* -n sighup -o jsonpath='{.spec.containers[0].image}'
+    ```
+
+8. Change the ```log.message.format.version``` of the Kafka resource to ```3.8.0```.
+
+   > Example: update ```log.message.format.version``` of ```kafka-cluster.yaml``` in ```examples/kafka-standard```.
+
+   Apply the changes:
+
+   ```kustomize build | kubectl apply -f -```
+
+9. Wait for the rollout of the Kafka resources, and check if the change has been correctly applied.
+
+   > Example command for ```kafka-cluster.yaml``` in ```examples/kafka-standard```.
+ 
+   ```kubectl get k sighup-kafka-cluster -n sighup -o jsonpath='{.spec.kafka.config.log\.message\.format\.version}'```
+
+   The version should be ```3.8.0```.
```

Comment:
The MAINTENANCE.md file is documentation for the maintainers (i.e. people that need to maintain the module, update it, etc.), not for end-users.

These instructions are awesome, but should not be here. Let's move them to a dedicated document (something like `docs/upgrade-guide-v0.1.0-tov0.2.0.md) so we can link it from the release notes when we'll release v0.2.0 (or the right versioning).

Also, these changes seem to be breaking, or at least require a lot of manual intervention, let's add a `[BREAKING]` to the PR title.